examples/stable_diffusion_v2/lora/lora.py

```diff
@@ -0,0 +1,234 @@
+from mindspore.ops import functional as F
+from mindspore.common.initializer import initializer, HeUniform
+# from tk.delta.delta_constants import VALID_TENSOR_DATATYPE
+from mindspore import dtype as mstype
+VALID_TENSOR_DATATYPE = [mstype.float16, mstype.float32]
+
+def is_version_ge(current_version, base_version):
+    """
+    description: Check whether the current version is higher than or equal to the base version.
+    args:
+        current_version: string, like "1.10.1"
+        base_version: string, like "2.0.0rc1"
+    return:
+        Bool: current_version >= base_version
+    example:
+        for current_version: 1.10.1, base_version: 2.0.0rc1, it return False.
+    """
+    version_split_char = '.'
+    if version_split_char not in base_version or version_split_char not in current_version:
+        raise ValueError("The version string will contain the `.`."
+                         "For example, current_version 1.8.1， base_version: 2.0.0.")
+    for x, y in zip(current_version.split(version_split_char), base_version.split(version_split_char)):
+        if not x.isdigit() or not y.isdigit():
+            continue
+        if int(x) != int(y):
+            return int(x) >= int(y)
+    return True
+
+if is_version_ge(ms.__version__, '2.0.0'):
+    import mindspore._checkparam as Validator
+    INC_LEFT = Validator.INC_LEFT
+else:
+    from mindspore._checkparam import Validator, Rel
+    INC_LEFT = Rel.INC_LEFT
+
+class LoRADense(nn.Dense):
+    """Define a dense layer with LoRA structure.
+
+    Attributes:
+        lora_in_channels (int): The number of channels in the input space.
+        lora_out_channels (int): The number of channels in the output space.
+        lora_rank(int): The number of rows(columns) in LoRA matrices.
+        lora_alpha(float): A constant in lora_rank.
+        lora_dropout(float): The dropout rate, greater equal than 0 and less than 1.
+        param_init_type(:class:`mindspore.dtype`): The type of data in initialized tensor.
+        compute_dtype(:class:`mindspore.dtype`): The compute type of data.
+
+    """
+
+    def __init__(
+            self,
+            in_channels: int,
+            out_channels: int,
+            lora_rank: int,
+            lora_alpha: float,
+            lora_dropout: float = 0.,
+            lora_a_init=HeUniform(negative_slope=math.sqrt(5)),
+            lora_b_init='zeros',
+            param_init_type=mstype.float32,
+            compute_dtype=mstype.float16,
+            **kwargs
+    ):
+        super(LoRADense, self).__init__(in_channels, out_channels, **kwargs)
+
+        # Check params
+        self._check_num(lora_rank, lora_alpha, lora_dropout)
+        self._check_init(lora_a_init, lora_b_init, lora_rank)
+        self._check_type_of_data(param_init_type, compute_dtype)
+
+        # Define and initialize params
+        self.lora_rank = lora_rank
+        self.lora_alpha = lora_alpha
+        if is_version_ge(ms.__version__, '2.0.0'):
```

Comment:
Referring the code in mindone, please change to  condition: ` version.parse(ms.__version__) > version.parse("1.10.1")`  for  operation API compatability

Reply:
ok